### PR TITLE
Add type annotations to torch.nn.modules.padding

### LINKED ISSUE #49492: Enable torch.nn.modules.padding typechecks during CI
`TestTypeHints.test_run_mypy` should pass with `[mypy-torch.nn.modules.padding]` deleted from mypy.ini

cc @ezyang @malfet @rgommers @xuzhao9 @gramster

### PR DESCRIPTION
**Pull request by guilhermeleobas**
Closes gh-49492

### CODE REVIEW

File: torch/nn/modules/padding.py
Comment (walterddr) [inline torch/nn/modules/padding.py:72]: good catch!

**Review by walterddr — APPROVED** [review]

**Review by rgommers — APPROVED** [review]

**Comment by rgommers** [comment]
One CI failure is unrelated.

File: torch/nn/modules/padding.py
Comment (malfet) [reply torch/nn/modules/padding.py:72]: Before merging the  change, please check that this code is still JIT compilable after the change and scalar to tuple promotions work as expected? (`_size_N_t` are special types which get automatic promotions from integer to list)
Requesting changes to just mention in the PR description which test covers that torch.nn.modules.padding is JITable after the change

**Review by malfet — CHANGES_REQUESTED** [review]
Please specify in comment what test cover that padding classes are `torch.jit.script`able after the change.

File: torch/nn/modules/padding.py
Comment (walterddr) [reply torch/nn/modules/padding.py:72]: actually in this particular class padding is always set by 
```
self.padding = _pair(padding)
```
which always return a tuple of 2

**Comment by guilhermeleobas** [comment]
Do not merge this PR until one checks if the annotations introduce any regression. See:
https://github.com/pytorch/pytorch/pull/49564#issuecomment-752916072

File: test/test_jit.py
Comment (walterddr) [inline test/test_jit.py:941]: was wondering if we can create list of tuple[nn.Module, Tensor] to simplify this. but we can probably do the refactoring in another pass

**Review by walterddr — APPROVED** [review]
looks good. let's wait for the test results

File: test/test_jit.py
Comment (guilhermeleobas) [reply test/test_jit.py:941]: Done!

**Review by malfet — APPROVED** [review]

**Comment by walterddr** [comment]
FYI on the failure for facebook internal not sure why the added padding test failed timeout internally. I am taking a look

File: test/test_jit.py
Comment (walterddr) [inline test/test_jit.py:948]: just checked and seems like this test is significantly longer comparing to other TestJIT items.
do we really need 320x480 size for the 3d replication padding test for JIT? or can we use a smaller size?

FYI I changed this to 32x48 and it passed internal test without timeout.

Comment (guilhermeleobas) [reply test/test_jit.py]: Done!

**Pull request closed without merging**
